File: tests/07_phases_ompss_v2.dg/c/success_task_01.c
/*
<testinfo>
test_generator="config/mercurium-ompss-v2"
test_nolink=yes
</testinfo>
*/

#pragma oss task
void f()
{ }

void g() {
    #pragma oss task
    {
        f();
    }

    #pragma oss task loop chunksize(4)
    for (int i = 0; i < 100; ++i)
    {}

    #pragma oss taskwait
}

File: tests/07_phases_ompss_v2.dg/fortran/success_task_01.f90
! <testinfo>
! test_generator=config/mercurium-ompss-v2
! test_nolink=yes
! </testinfo>

!$oss task
subroutine foo()
    implicit none

end subroutine foo

subroutine bar()
    implicit none
    integer :: i
    !$oss task
        call foo()
    !$oss end task

    !$oss task loop chunksize(4)
    do i=0, 100
    enddo
    !$oss end task

    !$oss taskwait
end subroutine bar
